MantidQt/CustomInterfaces/src/Indirect/ConvFit.cpp

```diff
@@ -279,29 +279,46 @@ void ConvFit::initFABADAOptions() {
 */
 void ConvFit::run() {
   // Get input from interface
-  const auto func = createFunction(m_uiForm.ckTieCentres->isChecked());
-  const auto function = std::string(func->asString());
   m_runMin = m_uiForm.spSpectraMin->value();
   m_runMax = m_uiForm.spSpectraMax->value();
   const auto specMin = m_uiForm.spSpectraMin->text().toStdString();
   const auto specMax = m_uiForm.spSpectraMax->text().toStdString();
+  auto cfs = sequentialFit(specMin, specMax, m_baseName);
+
+  // Add to batch alg runner and execute
+  m_batchAlgoRunner->addAlgorithm(cfs);
+  connect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this,
+          SLOT(sequentialFitComplete(bool)));
+  m_batchAlgoRunner->executeBatchAsync();
+}
+
+IAlgorithm_sptr ConvFit::sequentialFit(const std::string &specMin,
+                                       const std::string &specMax,
+                                       QString &outputWSName) {
```

Comment:
This function should modify the string that is passed to it. In the case of running the algorithm, with 'Run', this will change m_baseName. When performing a single fit, with 'Fit Single Spectrum', this will change m_singleFitOutputName.